### PR TITLE
Graded Exercises Lesson03 & Mailroom Part 1 (Assignment)

### PR DESCRIPTION
**Pull request by mglezglez**
Graded Exercises Lesson03 & Mailroom Part 1 (Assignment from Lesson03)

### CODE REVIEW

File: students/mgglez/lesson03/assignment03_mailroom_part1/mailroom.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            send_thank_you(donor_records)
+        elif choice == '2':
+            donor_records = {k: v for k, v in sorted(donor_records.items(), key=lambda item: sum(item[1]), reverse=True)}
+            generate_donor_report(donor_records)
+        elif choice == '3':
+            print('Goodbye!')
+            break
+
+if __name__ == "__main__":
+    main()
```
Comment (charlesrobison) [inline students/mgglez/lesson03/assignment03_mailroom_part1/mailroom.py:115]: Looks and works great - nice job on this

File: students/mgglez/lesson03/exercises_graded/list_lab.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    print(fruit_list)
+    # Ask the user for another fruit and add it to the end of the list.
+    another_fruit = input("Please, which other fruit do you want to add to this list? : ")
+    fruit_list.append(another_fruit)
+    # Display the list.
+    print(fruit_list)
+    # Ask the user for a number and display the number back to the user and the fruit corresponding
+    # to that number (on a 1-is-first basis). Remember that Python uses zero-based indexing, so you
+    # will need to correct.
+    f_index = int(input("Select the index for the fruit you want to display [1-{}] :".format(len(fruit_list))))
```
Comment (charlesrobison) [inline students/mgglez/lesson03/exercises_graded/list_lab.py:26]: Great!

File: students/mgglez/lesson03/exercises_graded/list_lab.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                while fruit in fruit_list:
+                    fruit_list.remove(fruit)
+                break
+            elif choice.lower().strip() == 'yes':
+                # Fruit will not be removed
+                break
+            else:
+                # Keep asking until a valid answer is provided
+                print("Your answer is invalid. Respond with 'yes' or 'no'")
+    print(fruit_list)
```
Comment (charlesrobison) [inline students/mgglez/lesson03/exercises_graded/list_lab.py:100]: Good!

**Review by charlesrobison — APPROVED** [review]
Nice job on these!

@natasha-aleksandrova - note there are a lot of extra files in this PR from other students - otherwise ready to merge

**Comment by natasha-aleksandrova** [comment]
@mglezglez can you please sync with upstream so that we can merge this?

please let us know if you need some help.

**Comment by mglezglez** [comment]
@natasha-aleksandrova what would I need to do to sync with upstream?

does that mean I need to do :

git pull upstream master 

Or do I need to do something else?

**Comment by mglezglez** [comment]
@natasha-aleksandrova  I had some issues with my local repository and had to recreate everything again. I was not able to sync my local repository from upstream. I created a new pull request with the exercises from Lesson 03 (same files you already reviewed) and the exercises for Lesson04 (pending review) https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1326 

Let me know if I can close this pull request 1319 and will leave https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/1326 opened .

**Comment by natasha-aleksandrova** [comment]
thank you! I will close this one.

**Pull request closed without merging**